clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -40,6 +39,7 @@
     private static final Logger log = LoggerFactory.getLogger(AbstractStickyAssignor.class);
 
     public static final int DEFAULT_GENERATION = -1;
+    public int maxGeneration = DEFAULT_GENERATION;
```

Comment:
put the `maxGeneration` into class scope, so we can re-use it in `prepopulateCurrentAssignments`.